crates/sdk/std/src/string/builder.rs

```diff
@@ -0,0 +1,50 @@
+use crate::ensure;
+use crate::{String, Token, Vec};
+
+/// Builds a [`String`] compliant with the Fixed-Gas rules.
+pub struct StringBuilder {
+    inner: Vec<u8>,
+}
+
+impl StringBuilder {
+    /// New builder, reserves room for `capacity` bytes.
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            inner: Vec::with_capacity(capacity),
+        }
+    }
+
+    /// Appends a [`String`] to the being built [`String`].
+    #[inline(never)]
+    pub fn push_str(&mut self, s: &String) {
+        let bytes = s.as_bytes();
+        ensure!(bytes.len() < 256);
+
+        seq_macro::seq!(N in 0..256 {
+            if N < bytes.len() {
+                let byte = bytes[N];
+                self.inner.push(byte);
+            }
+            else {
+                // halt immediately
+                return;
+            }
+        });
+    }
```

Comment:
Yes, but the most important things is to be Fixed-Gas compliant. The code size is was less important since a `Template` could be reused an infinite number of times.